### ISSUE TITLE
Feature request: Allow pP to be specified as a function of wind speed

### ISSUE THREAD

**Issue by paulf81** [NREL/floris#686]
# Allow pP to be specified as a function of wind speed

Currently in FLORIS, pP controls the rate of power loss to yaw by specifying an exponent.  In high wind speed, this reduction is automatically limited by converting to pW and effective wind speed.  However, it assumed to be fixed across low wind speeds.  However it could be useful to allow for pP to increase for example in lower wind speeds as observed in https://wes.copernicus.org/articles/6/1427/2021/

## Proposed solution
Allow for pP to be specified as a list with an accompanying list of wind speeds.  Challenge will be that users should take care not to double-count the effect of rated conditions on reducing apparent pP

## Alternatives considered
The approaches being piloted in Pull Request #638 might accomplish this intrinsically?

@ejsimley and @Bartdoekemeijer  curious what you think of this?

**Comment by Bartdoekemeijer** [comment]
I definitely like having more freedom to vary certain parameters. I do wonder whether this functionality may be better limited to the `time_series=True`, so that users can specify exact input parameters for:

1. Air density
2. Wind sheer
3. Turbulence intensity
4. pP

This way, you don't have to insert relationship tables inside of the FLORIS input file.

**Comment by paulf81** [comment]
That would be a more flexible approach.  To be honest now I'm not sure, I can imagine a few ways to implement this that have a set of pros and cons associated with each.  Some list like:

- Allow pP specified every time step
- pP provided over a list of wind speeds
- Formula for loss expanded to two coefficients to reflect yaw angle and wind speed dependency
- See if #638 handles this intrinsically

**Comment by misi9170** [comment]
Now that we have different operation models in v4, I think we can fairly clearly delineate:
- the [`CosineLossTurbine`](https://nrel.github.io/floris/operation_models_user.html#cosine-loss-model) uses a fixed loss exponent over all wind speeds (but as explained above, the effect "tapers off" at transition)
- More advanced turbine models (e.g. the [`ControllerDependentTurbine`](https://nrel.github.io/floris/operation_models_user.html#controller-dependent-model) and the [`UnifiedMomentumModel`](https://nrel.github.io/floris/operation_models_user.html#unified-momentum-model) can model power loss to yaw in different ways, which may well vary with wind speed.

@paulf81 , what do you think?

**Comment by paulf81** [comment]
I agree, we can also leave open a future model that implements a piece-wise empirical `pP`, but the hope is that [ControllerDependentTurbine](https://nrel.github.io/floris/operation_models_user.html#controller-dependent-model) and the [UnifiedMomentumModel](https://nrel.github.io/floris/operation_models_user.html#unified-momentum-model) take care of this more explicitly

**Comment by misi9170** [comment]
Ok, sounds good. For now I'm going to close this as incomplete, but we can reopen later if needed.
